xlr/sdk/src/validator.ts

```diff
@@ -173,15 +222,68 @@ export class XLRValidator {
     return validationIssues;
   }
 
+  private generateNestedTypesInfo(
+    potentialTypeErrors: Array<{
+      type: NodeType;
+      errors: Array<ValidationMessage>;
+    }>,
+    xlrNode: OrType,
+    rootNode: Node
+  ): { nestedTypesList: string; infoMessage?: string } {
+    const nestedTypes = new Set<string>();
+
+    potentialTypeErrors.forEach((typeError) => {
+      // Exclude so nested types don't try to resolve ExpressionRef | BindingRef
+      if (typeError.type.type !== "template") {
+        typeError.errors.forEach((error) => {
+          if (error.type === "type" && error.expected) {
+            // Split by separate types if union
+            String(error.expected)
+              .split(" | ")
+              .forEach((val) => nestedTypes.add(val.trim()));
+          }
+        });
+      }
+    });
+
+    // Display list of expected types as a union
+
+    let nestedTypesList =
+      Array.from(nestedTypes).slice(0, MAX_VALID_SHOWN).join(" | ") +
+      (nestedTypes.size > MAX_VALID_SHOWN
+        ? ` | ... +${nestedTypes.size - MAX_VALID_SHOWN}`
+        : "");
+
+    const docsURL = this.config.urlMapping;
+
+    // Support passing in a URL for matching type
+    if (docsURL && xlrNode.name && docsURL[xlrNode.name]) {
+      nestedTypesList = docsURL[xlrNode.name];
+    }
+
+    // Support supplemental info message
+    let infoMessage;
+
+    if (rootNode.value !== undefined) {
+      infoMessage = `Got: ${rootNode.value} and expected: ${nestedTypesList}\n`;
+    } else if (nestedTypesList) {
+      infoMessage = `Expected: ${nestedTypesList} | (binding/expression)`;
```

Comment:
I think this logic might be causing a weird issue resulting in the error message [here](https://github.com/player-ui/tools/pull/189/files#diff-b8e1487d86ecbc9a5592444155e7f170ae2784acd2856b714b9fa0442729b098R60) since `AssetWrapperOrSwitch` should just be an array of types (AssetWrapper, StaticSwitch, and DynamicSwitch) that are checked, and not `binding` nor `expression`.

Reply:
Oh shoot, ok let me troubleshoot this